CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/SingleChoiceInputPane.java

```diff
@@ -115,42 +115,42 @@ public SingleChoiceInputPane(final PluginParameter<SingleChoiceParameterValue> p
             SingleChoiceParameterType.setChoiceData(parameter, field.getSelectionModel().getSelectedItem());
         });
 
-        parameter.addListener((final PluginParameter<?> parameter1, final ParameterChange change) -> {
+        parameter.addListener((final PluginParameter<?> scParameter, final ParameterChange change) -> {
             Platform.runLater(() -> {
-                @SuppressWarnings("unchecked") //scParameter1 is SingleChoiceParameter
-                PluginParameter<SingleChoiceParameterValue> scParameter1 = (PluginParameter<SingleChoiceParameterValue>) parameter1;
+                assert(scParameter.getParameterValue() instanceof SingleChoiceParameterValue);
+                SingleChoiceParameterValue scParameterValue = (SingleChoiceParameterValue) scParameter.getParameterValue();
```

Comment:
final

Reply:
So final. The final final.

Thanks for being so thorougher in your review! Evidently I need to pay more attention to when things can be final.

Reply:
No worries. Evidently so do I else I wouldn't need to do more than one review :)